src/instruction.rs

```diff
@@ -116,6 +121,55 @@ pub enum Instruction {
         alternative: Option<Vec<Instruction>>,
     },
 }
+impl PartialEq for Instruction {
```

Comment:
This couldn't be derived?

Reply:
You are right! Im sure it was complaining about `Rc` or something else not implementing `PartialEq`. But now when I tried it it's fine :exploding_head: 
I'll rewrite :)

Reply:
removed :point_right: 21b930426c79df8edf415613756b64514b04da4e